test/OpenTelemetry.Tests/Metrics/MultipleReadersTests.cs

```diff
@@ -154,6 +153,51 @@ public void SdkSupportsMultipleReaders(AggregationTemporality aggregationTempora
             }
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void ObservableInstrumentCallbacksInvokedForEachReaders(bool hasViews)
+        {
+            var exportedItems1 = new List<Metric>();
+            var exportedItems2 = new List<Metric>();
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{hasViews}");
+            int callbackInvocationCount = 0;
+            var gauge = meter.CreateObservableGauge("gauge", () =>
+            {
+                callbackInvocationCount++;
+                return 10 * callbackInvocationCount;
+            });
+
+            var meterProviderBuilder = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(exportedItems1)
+                .AddInMemoryExporter(exportedItems2);
+
+            if (hasViews)
+            {
+                meterProviderBuilder.AddView("gauge", "renamedGauge");
+            }
+
+            using var meterProvider = meterProviderBuilder.Build();
+            meterProvider.ForceFlush();
+
+            // VALIDATE
+            Assert.Equal(2, callbackInvocationCount);
+            Assert.Single(exportedItems1);
+            Assert.Single(exportedItems2);
```

Comment:
Add an `Assert` to test the `MetricPoint` value as well?

Reply:
saving for a future PR. This has plenty of improvements possible. The current test does not test the case where each reader is on its own interval. The spec has clear requirements on one reader not affecting others.. Need to improve this area.

Reply:
Need to improve this area  --> Need to improve tests in this area